.github/workflows/test_rust.yml

```diff
@@ -153,6 +156,49 @@ jobs:
         env:
           RUSTDOCFLAGS: -D warnings
 
+  coverage:
+    needs: changes
+    if: needs.changes.outputs.should_run == 'true'
+    name: Coverage Report
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: actions/checkout@v5
+
+      - name: Install Rust toolchain
+        # We need nightly for coverage due to:
+        #  1. wayland-backend requiring that,
+        #  2. branch coverage being unstable.
+        #
+        # Ad 1.
+        #   See <https://github.com/Smithay/wayland-rs/commit/05d8307ea647053b949357eb8588dfdab394f4fe>
+        # Ad 2.
+        #   See <https://github.com/taiki-e/cargo-llvm-cov/issues/8>
+        #   See <https://github.com/rust-lang/rust/issues/79649>
+        uses: dtolnay/rust-toolchain@nightly
+        with:
+          components: llvm-tools
+
+      - name: Install llvm-cov
+        uses: taiki-e/install-action@v2
+        with:
+          tool: cargo-llvm-cov
+
+      - name: Install Linux dependencies
+        run: |
+          sudo apt-get update
+          sudo apt install -y ${{ env.LINUX_DEPENDENCIES }}
+
+      - name: Run llvm-cov for SWF tests
+        working-directory: tests
+        run: cargo llvm-cov --html ${{ env.LLVM_COV_OPTS }}
```

Comment:
nit: Why is `--html` not in the envvar?

Reply:
That's because it's a local option, not global. You can call `llvm-cov` in a different way, e.g. first generate coverage without the report and then generate the report separately. Global options wouldn't change then, but `--html` would.